importer.py

```diff
@@ -21,4 +21,37 @@
                              attachment_key='org.digitalpsych.redcap.data',
                              body=records)
     print(count)
+    try:
+        arr = LAMP.Type.get_attachment(researcher['id'], 'org.digitalpsych.redcap.links')["data"]
+    except LAMP.ApiException:
+        continue # Ignore this Researcher
+    parts = []
+    for study in LAMP.Study.all_by_researcher(researcher['id'])['data']:
+        parts+=(p['id'] for p in LAMP.Participant.all_by_study(study['id'])['data'])
+    for p in parts:
+        try:
+            record_id = LAMP.Type.get_attachment(p, 'org.digitalpsych.redcap.id')["data"]
+        except LAMP.ApiException:
+            continue # Ignore this Participant
+        if record_id >= 0:
+            try:
+                link = LAMP.Type.get_attachment(p, 'org.digitalpsych.redcap.links')["data"]
+            except LAMP.ApiException:
+                links = {}
+                for s in arr:
+                    data = {
+                        'token': config['API_TOKEN'],
+                        'content': 'surveyLink',
+                        'format': 'json',
+                        'instrument': s,
+                        'event': '',
+                        'record': record_id,
+                        'returnFormat': 'json'
+                    }
+                    r = requests.post(config['API_URL'], data=data)
+                    links[s] = r.text
```

Comment:
Is this a text list of each link or something? Can we make sure to at least format it into a JSON array if possible?

Reply:
I'm not sure why the API returns it this way. But I'm putting it into a dict to add to the tag.

Reply:
Makes sense! 👍  The RedCap API is pretty barebones.